sdks/go/pkg/beam/core/runtime/graphx/schema/logicaltypes.go

```diff
@@ -0,0 +1,122 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/apache/beam/sdks/go/pkg/beam/core/util/reflectx"
+)
+
+var (
+	// Maps logical type identifiers to their reflect.Type and the schema representation.
+	// the type identifier is the reflect.Type name, and included in the proto as well.
+	// We don't treat all types as "logical" types.
+	// ... why don't we treat all types as Logical types?
+	logicalTypes       = map[string]LogicalType{}
+	logicalIdentifiers = map[reflect.Type]string{}
```

Comment:
I don't understand the reason to have the ID layer between logicalIdentifiers and logicalTypes. Is there a reason not to just have this?:
```suggestion
	logicalTypes       = map[reflect.Type]LogicalType{}
```

Reply:
Good question: the string lookup is because that's all we have from the Schema proto direction. We need a mapping to the Logical part regardless, and then we need a mapping from the reflect.Type to that identifier. There's no guarantee or reason that a logical type's identifier will actually be related to the reflect.Type information unfortunately.

Reply:
Ack, that makes sense. I hadn't considered what to do when coming from the proto direction.